instrumentation/CONTRIBUTING.md

```diff
@@ -0,0 +1,483 @@
+
+### Use Semantic Conventions
+
+Use the OpenTelemetry Semantic Conventions to ensure that the instrumentation is compatible with other OpenTelemetry libraries and that the data is useful in a distributed context.
+
+> :information_source: Privacy and security are important considerations when adding attributes to spans. Please ensure that you are not adding sensitive information to spans. If you are unsure, please ask for a review.
+
+When semantic conventions do not exist, use the [Elastic Common Schema](https://www.elastic.co/guide/en/ecs/current/index.html) and submit a Issue/PR with your attributes to the [Semantic Conventions repo](https://github.com/open-telemetry/semantic-conventions) to propose a new set of standard attributes.
+
+If the attribute is specific to your instrumentation, then consider namespacing it using the `instrumentation` prefix e.g. `werewolf.bite.damage`.
+
+### Write Comprehensive Automated Tests
+
+Code that is not tested will not be accepted by maintainers. We understand that providing 100% test coverage is not always possible but we still ask that you provide your best effort when writing automated tests.
+
+Most of the libraries we are instrumenting introduce changes that are outside of our control. For this reason integration or state based tests are preferred over interaction (mock) tests.
+
+When you do in fact run into cases where test doubles or API stubs are absolutely necessary then we recommend using the `rspec-mocks` and `webmocks` gems.
+
+### Understand Performance Characteristics
+
+The OTel Specification describes expectations around the performance of SDKs, which you must review and apply to instrumentations: <https://opentelemetry.io/docs/specs/otel/performance/>
+
+Instrumentation libraries should be as lightweight as possible and must:
+
+* Rely on `rubocop-performance` linters to catch performance issues
+* Consider using [microbenchmarks](https://github.com/evanphx/benchmark-ips) and [profiling](https://ruby-prof.github.io/) to address any possible performance issues
+* Provide minimal solutions and code paths
+
+#### Provide minimal solutions and code paths
+
+Instrumentations should have the minimal amount of code necessary to provide useful insights to our users. It may sound contrary to good engineering practices, but you must avoid adding lots of small methods, classes, and objects when instrumenting a library.
+
+Though often easier to maintain and reason about; small and well factored code adds overhead to the library you are instrumenting resulting in performance degradation due to unnecessary object allocations, method dispatching, and other performance overhead.
+
+It will also contribute to building large backtraces that makes it more difficult for our end users to understand the essential parts of exception reports application. That will in turn likely result in additional filtering logic in the their application to avoid reporting unnecessary stack frames.
```

Comment:
```suggestion
It also contributes to building large backtraces, making it more difficult for our end users to understand the essential parts of exception reports. That will likely result in additional filtering logic in their application to avoid reporting unnecessary stack frames.
```